syntax = "proto3";

package smartcity;

message DeviceAnnounce {
  string device_id = 1;
  string device_type = 2;
  string ip = 3;
  int32 port_udp = 4;
  int32 port_tcp = 5;
}

message DeviceInfo {
  string device_id = 1;
  string device_type = 2;
  string ip = 3;
  int32 port_udp = 4;
  int32 port_tcp = 5;
  double last_value = 6;
  int64 last_seen = 7;
}

message GatewayRequest {
  string action = 1;        // "LIST"
}

message GatewayResponse {
  bool ok = 1;
  repeated DeviceInfo devices = 2;
  string msg = 3;
}
